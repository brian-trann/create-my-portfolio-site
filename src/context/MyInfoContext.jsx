import React from 'react';
const MyInfoContext = React.createContext();
export default MyInfoContext;
