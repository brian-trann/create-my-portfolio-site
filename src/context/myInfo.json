{
    "navigation": {
        "name": "John Doe",
        "menu": [
            {
                "text": "Home",
                "url": "/",
                "target":""
            },
            {
                "text": "About",
                "url": "/about",
                "target":""
            },
            {
                "text": "Github",
                "url": "https://github.com/",
                "target":"_blank"
            },
            {
                "text": "LinkedIn",
                "url": "https://www.linkedin.com//",
                "target":"_blank"
            }
        ]
    },
    "main": {
        "greeting": "Hi, I'm John, a Software Developer from California",
        "blurb":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "profileImage": { 
            "url":"",
            "altText":"profile image"
        },
        "image": {
            "url": "/",
            "altText": "temp"
        }
    },
    "work":{
        "categoryName":"Selected Work",
        "projects":[
            {
                "projectName": "project0",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project0 thumbnail"
                }
            },
            {
                "projectName": "project1",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project1 thumbnail"
                }
            },
            {
                "projectName": "project2",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project2 thumbnail"
                }
            }
        ],
        "buttonNames":{
            "primary":"Github",
            "secondary":"Site"
        }
    },
    "about": {
        "categoryName":"About",
        "statement": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "skills":{
            "categoryName":"Skills",
            "items":[
                {
                    "type":"Languages",
                    "items":["JavaScript", "HTML", "CSS","Python","SQL"]
                },
                {
                    "type":"Libraries and Frameworks",
                    "items":["React", "Redux", "Express","Flask"]
                },
                {
                    "type":"Other",
                    "items":["other0", "other1"]
                }
            ]
        },
        "workExperience": {
            "categoryName":"Work Experience",
            "work": [
                {
                    "company":"company0",
                    "position":"position0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                },
                {
                    "company":"company2",
                    "position":"position1",
                    "date":"Date 2019 - Date 2020",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "education":{
            "categoryName":"Education",
            "items": [
                {
                    "name":"education0",
                    "curriculum":"curriculum0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "certifications":{
            "categoryName":"Certifications",
            "items":["certification0", "certification1"]
        }
    },
    "other":{
        "images":[
            {"image":{
                "url":"https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                "altText":"",
                "description":""
                }
            }
        ]
    }
}