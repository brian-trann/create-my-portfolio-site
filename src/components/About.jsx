import React, { useContext } from 'react';
import MyInfoContext from '../context/MyInfoContext';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Divider from '@material-ui/core/Divider';
import Paper from '@material-ui/core/Paper';
import CardMedia from '@material-ui/core/CardMedia';
import Card from '@material-ui/core/Card';

const useStyles = makeStyles((theme) => ({
	container        : {
		padding : theme.spacing(6, 0, 6)
	},
	category         : {
		padding : theme.spacing(4)
	},
	categoryBody     : {
		paddingTop  : theme.spacing(2),
		paddingLeft : theme.spacing(2)
	},
	aboutImage       : {
		maxWidth : '700px',
		display  : 'block',
		margin   : 'auto'
	},
	media            : {
		height : 140
	},
	mediaDescription : {
		paddingTop : theme.spacing(2)
	},
	aboutStatement   : {
		paddingTop    : theme.spacing(1),
		paddingBottom : theme.spacing(4)
	}
}));
const About = () => {
	const { about, other: { images } } = useContext(MyInfoContext);

	const { statement, skills, workExperience, education, certifications } = about;
	const classes = useStyles();
	return (
		<React.Fragment>
			<Container maxWidth='md' className={classes.container}>
				<Paper elevation={2}>
					{/* About */}
					<div className={classes.category}>
						<Typography variant='h4'>{about.categoryName}</Typography>
						<div className={classes.categoryBody}>
							<Typography variant='subtitle1' className={classes.aboutStatement}>
								{statement}
							</Typography>
						</div>

						{/* About Image */}
						{images.length > 0 && (
							<React.Fragment>
								<Card className={classes.aboutImage}>
									<CardMedia
										image={images[0].image.url}
										title={images[0].image.altText}
										className={classes.media}
									/>
								</Card>
								<Typography
									color='textSecondary'
									align='center'
									display='block'
									variant='caption'
									className={classes.mediaDescription}
								>
									{images[0].image.description}
								</Typography>
							</React.Fragment>
						)}
						{/* End About Image */}
					</div>
					<Container maxWidth='sm'>
						<Divider />
					</Container>
					{/* End About */}
					{/* Skills */}
					<div className={classes.category}>
						<Typography variant='h4'>{skills.categoryName}</Typography>
						<div className={classes.categoryBody}>
							{skills.items.map((skill, i) => (
								<div className={i} key={i}>
									<Typography variant='h6' gutterBottom>
										{skill.type}
									</Typography>
									<Typography>{skill.items.join(', ')}</Typography>
								</div>
							))}
						</div>
					</div>
					{/* EndSkills */}
					<Container maxWidth='sm'>
						<Divider />
					</Container>
					{/* Work Experience */}
					<div className={classes.category}>
						<Typography variant='h4'>{workExperience.categoryName}</Typography>
						<div className={classes.categoryBody}>
							{workExperience.work.map((w) => (
								<div key={w.company}>
									<Typography variant='h6' gutterBottom>
										{w.company}
									</Typography>
									<Typography gutterBottom>{w.position}</Typography>
									<Typography gutterBottom>{w.date}</Typography>
									<Typography gutterBottom>{w.description}</Typography>
								</div>
							))}
						</div>
					</div>
					{/* End Work Experience */}
					<Container maxWidth='sm'>
						<Divider />
					</Container>
					{/* Education*/}
					<div className={classes.category}>
						<Typography variant='h4'>{education.categoryName}</Typography>
						<div className={classes.categoryBody}>
							{education.items.map((e) => (
								<div key={e.name}>
									<Typography variant='h6' gutterBottom>
										{e.name}
									</Typography>
									<Typography gutterBottom>{e.curriculum}</Typography>
									<Typography gutterBottom>{e.date}</Typography>
									<Typography gutterBottom>{e.description}</Typography>
								</div>
							))}
						</div>
					</div>
					{/* End Education*/}
					<Container maxWidth='sm'>
						<Divider />
					</Container>
					{/* Certifications*/}
					<div className={classes.category}>
						<Typography variant='h4'>{certifications.categoryName}</Typography>
						<div className={classes.categoryBody}>
							<Typography>{certifications.items.join(', ')}</Typography>
						</div>
					</div>
					{/* End Certifications*/}
				</Paper>
			</Container>
		</React.Fragment>
	);
};
export default About;
