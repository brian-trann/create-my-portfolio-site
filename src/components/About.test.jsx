import React from 'react';
import { render } from '@testing-library/react';
import About from './About';
import myInfo from '../context/myInfo.json';
import MyInfoContext from '../context/MyInfoContext';

// smoke test
it('renders without crashing', () => {
	render(
		<MyInfoContext.Provider value={myInfo}>
			<About />
		</MyInfoContext.Provider>
	);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(
		<MyInfoContext.Provider value={myInfo}>
			<About />
		</MyInfoContext.Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
