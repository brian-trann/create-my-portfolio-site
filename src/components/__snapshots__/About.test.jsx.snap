// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-container-8 MuiContainer-maxWidthMd"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="makeStyles-category-9"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          About
        </h4>
        <div
          class="makeStyles-categoryBody-10"
        >
          <h6
            class="MuiTypography-root makeStyles-aboutStatement-14 MuiTypography-subtitle1"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
          </h6>
        </div>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-aboutImage-11 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardMedia-root makeStyles-media-12"
            style="background-image: url(https://github.githubassets.com/images/modules/site/home/repo-browser.png);"
            title=""
          />
        </div>
        <span
          class="MuiTypography-root makeStyles-mediaDescription-13 MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-alignCenter MuiTypography-displayBlock"
        />
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div
        class="makeStyles-category-9"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Skills
        </h4>
        <div
          class="makeStyles-categoryBody-10"
        >
          <div
            class="0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Languages
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              JavaScript, HTML, CSS, Python, SQL
            </p>
          </div>
          <div
            class="1"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Libraries and Frameworks
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              React, Redux, Express, Flask
            </p>
          </div>
          <div
            class="2"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Other
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              other0, other1
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div
        class="makeStyles-category-9"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Work Experience
        </h4>
        <div
          class="makeStyles-categoryBody-10"
        >
          <div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              company0
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              position0
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Date 2020 - Date 2021
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
          <div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              company2
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              position1
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Date 2019 - Date 2020
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div
        class="makeStyles-category-9"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Education
        </h4>
        <div
          class="makeStyles-categoryBody-10"
        >
          <div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
            >
              education0
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              curriculum0
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Date 2020 - Date 2021
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div
        class="makeStyles-category-9"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Certifications
        </h4>
        <div
          class="makeStyles-categoryBody-10"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            certification0, certification1
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
